src/Analysis/Ast/Impl/Modules/PythonModule.cs

```diff
@@ -352,18 +353,22 @@ internal PythonModule(ModuleCreationOptions creationOptions, IServiceContainer s
         }
 
         private void Parse(CancellationToken cancellationToken) {
-            var sink = new CollectingErrorSink();
+            CollectingErrorSink sink = null;
             int version;
             Parser parser;
 
             //Log?.Log(TraceEventType.Verbose, $"Parse begins: {Name}");
 
             lock (AnalysisLock) {
                 version = _buffer.Version;
-                parser = Parser.CreateParser(new StringReader(_buffer.Text), Interpreter.LanguageVersion, new ParserOptions {
-                    StubFile = FilePath != null && Path.GetExtension(FilePath).Equals(".pyi", FileSystem.StringComparison),
-                    ErrorSink = sink
-                });
+                var options = new ParserOptions {
+                    StubFile = FilePath != null && Path.GetExtension(FilePath).Equals(".pyi", FileSystem.StringComparison)
+                };
+                if (ModuleType == ModuleType.User) {
```

Comment:
The only downside of this is that if there's a syntax error or other diagnostic in a library file and the user does a go-to-def, nothing will show up (since the module type isn't user, so errors/etc are not stores). I'm not sure how critical that is, though.

Reply:
This only affects error collection, the library goto def should still work, provided library is in source.

Reply:
I mean that the newly opened non-user file will not show any diagnostics because they were never collected, not that go-to-def won't work.

Reply:
Yes, that is intentional. Diagnostics would be noise for the user since this is not something that is actionable for them.